src/vcfio/utils/str_utils.py

```diff
@@ -33,12 +31,46 @@ def standardize_chromosome(raw_chr):
     return chromosome
 
 
-def calculate_zygosity(gt: AnyStr) -> Zygosity:
+def calculate_zygosity(genotype: str) -> Optional[Zygosity]:
     """
     calculate zygosity of a sample
     examples:
         '0/1' -> HET
         './.' -> NO_COVERAGE
     """
-    gt_pair = GT_PATTERN.findall(gt)[0]
-    return GT_ZYGOSITY_MAP.get(gt_pair, Zygosity.no_coverage)
\ No newline at end of file
+    zygosity = None
+    if len(genotype) == 1:
+        zygosity = _parse_haploid(genotype)
+    elif len(genotype) == 3:
+        zygosity = _parse_diploid(genotype)
+
+    return zygosity
+
+
+def _parse_haploid(genotype: str) -> Optional[Zygosity]:
+    if genotype == '0':
+        return Zygosity.reference
+    elif genotype.isdigit():
+        return Zygosity.homozygote
+    else:
+        return None
```

Comment:
According to VCF specs, a dot is a possible value in which case we should return Zygosity.no_coverage.

https://samtools.github.io/hts-specs/VCFv4.3.pdf (page 11, at the top)

Reply:
I wonder how the `chrM	726` test passes if a dot will result into a None which should result into an exception?

Reply:
We indeed return no_coverage in this case.
I also added a unit test to check this.
A single dot is considered an empty value, so we don't get here.
But we still return no_coverage.

Reply:
A single dot is an empty value, not a haploid GT, so we don't get to this function.
The caller, get_zygosity() replaces empty values with './.'

Reply:
Understood, thanks